enos/README.md

```diff
@@ -0,0 +1,259 @@
+
+```bash
+# List all available scenarios
+enos scenario list
+# Run the smoke or upgrade scenario with an artifact that is built locally. Make sure
+# the local machine has been configured as detailed in the requirements
+# section. This will execute the scenario and clean up any resources if successful.
+enos scenario run smoke artifact_source:local
+enos scenario run upgrade artifact_source:local
+# To run the same scenario variants that are run in CI, refer to the scenarios listed
+# in json files under .github/enos-run-matrices directory,
+# adding `artifact_source:local` to run locally.
+enos scenario run smoke backend:consul consul_version:1.12.3 distro:ubuntu seal:awskms artifact_source:local arch:amd64 edition:oss
+# Launch an individual scenario but leave infrastructure up after execution
+enos scenario launch smoke artifact_source:local
+# Check an individual scenario for validity. This is useful during scenario
+# authoring and debugging.
+enos scenario validate smoke artifact_source:local
+# If you've run the tests and desire to see the outputs, such as the URL or
+# credentials, you can run the output command to see them. Please note that
+# after "run" or destroy there will be no "outputs" as the infrastructure
+# will have been destroyed and state cleared.
+enos scenario output smoke artifact_source:local
+# Explicitly destroy all existing infrastructure
+enos scenario destroy smoke artifact_source:local
+```
+
+Refer to the [Enos documentation](https://github.com/hashicorp/Enos-Docs)
+for further information regarding installation, execution or composing scenarios.
+
+## UI Tests
+The [`ui` scenario](./enos-scenario-ui.hcl) creates a Vault cluster (deployed to AWS) using a version 
+built from the current checkout of the project. Once the cluster is available the UI acceptance tests
+are run in a headless browser.
+### Variables
+In addition to the required variables that must be set, as described in the [Scenario Variables](#Scenario Variables), 
+the `ui` scenario has two optional variables:
+
+**ui_test_filter** - An optional test filter to limit the tests that are run, i.e. `'!enterprise'`.
+To set a filter export the variable as follows:
+```shell
+> export ENOS_VAR_ui_test_filter="some filter"
+```
+**ui_run_tests** - An optional boolean variable to run or not run the tests. The default value is true. 
+Setting this value to false is useful in the case where you want to create a cluster, but run the tests 
+manually. The section [Running the Tests](#Running the Tests) describes the different ways to run the
+'UI' acceptance tests.
+
+### Running the Tests
+The UI tests can be run fully automated or manually.
+#### Fully Automated
+The following will deploy the cluster, run the tests, and subsequently tear down the cluster: 
+```shell
+> export ENOS_VAR_ui_test_filter="some filter" # <-- optional
+> cd enos
+> enos scenario ui run edition:oss
+```
+#### Manually
+The UI tests can be run manually as follows:
```

Comment:
Let's remove anything in this README that doesn't apply to LDAP. You can do that in a follow-up PR if you prefer.